### ISSUE TITLE
[Bug] universe.dagger.io/yarn cache mount config results in yarn package corruption

### ISSUE THREAD

**Issue by gerhard** [dagger/dagger#1670]
## What is the problem?

Running `yarn.#Build` & `yarn.#Run` in parallel always results in yarn package corruption:

```sh
5:13PM INF actions.test.container._exec | computing
5:13PM INF actions.build.container._exec | computing
5:13PM INF actions.build.container._exec | #9 1.958 yarn install v1.22.17
5:13PM INF actions.build.container._exec | #9 2.751 [1/4] Resolving packages...
5:13PM INF actions.build.container._exec | #9 6.398 [2/4] Fetching packages...
5:13PM INF actions.build.container._exec | #9 9.576 error https://registry.yarnpkg.com/supports-color/-/supports-color-5.5.0.tgz: Integrity check failed for "supports-color" (computed integrity doesn't match our records, got "sha512-z4PhNX7vuL3xVChQ1m2AB9Yg5AULVxXcg/SpIdNs6c5H0NE8XYXysP+DGNKHfuwvY7kxvUdBeoGlODJ6+SfaPg== sha1-2jmj7l5rSw0yVb/vlWAYkK/YBwk=")
5:13PM INF actions.build.container._exec | #9 9.577 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
5:13PM INF actions.test.container._exec | #8 2.006 yarn install v1.22.17
5:13PM INF actions.test.container._exec | #8 2.784 [1/4] Resolving packages...
5:13PM INF actions.test.container._exec | #8 6.451 [2/4] Fetching packages...
5:13PM INF actions.test.container._exec | #8 9.592 error An unexpected error occurred: "ENOENT: no such file or directory, open '/cache/yarn/v6/npm-supports-color-5.5.0-e2e69a44ac8772f78a1ec0b35b689df6530efc8f-integrity/node_modules/supports-color/.yarn-metadata.json'".
5:13PM INF actions.test.container._exec | #8 9.593 info If you think this is a bug, please open a bug report with the information provided in "/src/yarn-error.log".
5:13PM INF actions.test.container._exec | #8 9.593 info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
5:13PM ERR actions.build.container._exec | failed: process "bash --norc -e -o pipefail /bash/scripts/run.sh" did not complete successfully: exit code: 1    duration=18.1s
5:13PM ERR actions.test.container._exec | canceled    duration=18.1s
5:13PM FTL system | failed to up environment: task failed: actions.build.container._exec: process "bash --norc -e -o pipefail /bash/scripts/run.sh" did not complete successfully: exit code: 1
```

## How to reproduce it?

Follow https://deploy-preview-1586--devel-docs-dagger-io.netlify.app/1200/local-dev/ with a dev version of `dagger`.

## Did this use to work?

Yes, [11 days ago](https://github.com/dagger/examples/pull/54#issuecomment-1045298941), with `v0.2.0-alpha.5`.

I suspect [buildkit v0.10.0-rc.2](https://github.com/moby/buildkit/releases/tag/v0.10.0-rc2) may have something to do with it, but it's just a guess.

## What should happen next?

Find the simplest fix.

By the way, trying to fix this problem by setting the `name` property triggers other bugs. I time-boxed it to 30 mins and having gone in circles, I moved it to out of scope.

cc @samalba @aluzzardi @shykes 

---

This was discovered part of https://github.com/dagger/dagger/pull/1586.
It is blocking the most important Europa doc: **Getting Started - CI/CD in your local dev**
